cs/cs/Serializer/VowpalWabbitJsonSerializer.cs

```diff
@@ -91,82 +103,130 @@ public VowpalWabbitExample Parse(string json, ILabel label = null)
+                                    //    break;
+                                    //case "_multi":
+                                    //    break;
+                                    default:
+                                        reader.Skip();
+                                        break;
                                 }
+                            }
+                            else
+                            {
+                                if (!reader.Read())
+                                    throw new VowpalWabbitJsonException(reader.Path, "Unexpected end");
+
+                                if (reader.TokenType == JsonToken.StartObject)
+                                    this.ParseNamespaceAndFeatures(this.Context, propertyName);
                                 else
-                                {
-                                    if (!reader.Read())
-                                        throw new VowpalWabbitJsonException(reader.Path, "Unexpected end");
-
-                                    if (reader.TokenType == JsonToken.StartObject)
-                                        this.ParseNamespaceAndFeatures(context, propertyName);
-                                    else
-                                        this.ParseFeature(defaultNamespaceContext, defaultNamespace, propertyName);
+                                    this.ParseFeature(this.DefaultNamespaceContext, defaultNamespace, propertyName);
 
-                                }
-                                break;
-                        }
+                            }
+                            break;
+                        case JsonToken.EndObject:
+                            goto done;
                     }
+                }
+            done:
 
-                    // append default namespaces features if we found some
-                    if (defaultNamespaceContext.StringExample != null && defaultNamespaceContext.StringExample.Length > 0)
-                    {
-                        context.StringExample.AppendFormat(CultureInfo.InvariantCulture,
-                            "| {0}", defaultNamespaceContext.StringExample);
-                    }
+                // append default namespaces features if we found some
+                if (this.DefaultNamespaceContext.StringExample != null && this.DefaultNamespaceContext.StringExample.Length > 0)
+                {
+                    this.Context.StringExample.AppendFormat(CultureInfo.InvariantCulture,
+                        " | {0}", this.DefaultNamespaceContext.StringExample);
                 }
+            }
+        }
 
-                var vwExample = context.ExampleBuilder.CreateExample();
+        /// <summary>
+        /// Creates the managed example representation.
+        /// </summary>
+        /// <returns>Returns the managed example.</returns>
+        public VowpalWabbitExample CreateExample()
+        {
+            try
```

Comment:
You need initialize in the using(…), and the contexts are initialized in the ctor.